components/renku_data_services/notebooks/utils.py

```diff
@@ -128,6 +130,21 @@ def node_affinity_from_resource_class(
     return affinity
 
 
+def node_affinity_patch_from_resource_class(
+    resource_class: crc_models.ResourceClass, default_affinity: Affinity
+) -> AffinityPatch:
+    """Create a patch for the session affinity."""
+    affinity = node_affinity_from_resource_class(resource_class, default_affinity)
+    patch = AffinityPatch(nodeAffinity=RESET, podAffinity=RESET, podAntiAffinity=RESET)
+    if affinity.nodeAffinity:
+        patch.nodeAffinity = affinity.nodeAffinity
```

Comment:
Node affinity is not cleared when the service is deployed without a default node affinity.

Reply:
Tested here: https://github.com/SwissDataScienceCenter/renku/pull/4235

Reply:
I propagated the affinity patching all the way through now. So it should work. I will test it tomorrow.
see https://github.com/SwissDataScienceCenter/renku-data-services/pull/1059/commits/08e3941556960c2cba99e2f3bacad8b46358c72a